src/Resources/views/CRUD/base_list.html.twig

```diff
@@ -118,7 +118,7 @@ file that was distributed with this source code.
                                         {# NEXT_MAJOR: remove the attribute check and just use .countResults() #}
                                         {%- if (attribute(admin.datagrid.pager, 'countResults') is defined ? admin.datagrid.pager.countResults() : admin.datagrid.pager.getNbResults()) > 0 -%}
                                             <li>
-                                                <a href="{{ admin.generateUrl('list', {'filter': admin.datagrid.values|merge({(constant('Sonata\\AdminBundle\\Datagrid\\DatagridInterface::PAGE')): 1})}) }}">
+                                                <a href="{{ admin.generateUrl('list', {'filter': admin.datagrid.getPaginationParameters(1)}) }}">
```

Comment:
`DataGrid::getPaginationParameters` returns some nested array with a `filter` key? This seems wrong then to me

https://github.com/sonata-project/SonataAdminBundle/blob/3.x/src/Datagrid/Datagrid.php#L310

Reply:
Indeed, it might be better now.

Reply:
Just need to fix
```
Uncaught PHP Exception Twig\Error\RuntimeError: "The merge filter only works with arrays or "Traversable", got "NULL" as first argument." at /Users/vincentl/Perso/SonataAdminBundle/src/Resources/views/Pager/base_results.html.twig line 28
```
^^